### ISSUE TITLE
Inheritdoc XML documentation should not self-reference

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#326]
The XML documentation via the `inheritdoc` tag should not reference the type itself.

E.g. following code should report a violation:
```c#
    /// <inheritdoc cref="Sample" />
    public class Sample
    { }
```
